internal/fields/mappings.go

```diff
@@ -624,22 +687,295 @@ func (v *MappingValidator) validateMappingsNotInPreview(currentPath string, chil
+			// Force to match beginning and ending
+			r = fmt.Sprintf("^%s$", r)
+
+			regexes[i] = r
+		}
+		return stringMatchesPatterns(regexes, elem)
+	}
+
+	fieldType := mappingParameter("type", definition)
+	if fieldType == "" {
+		return fmt.Errorf("missing type parameter for field: %q", currentPath)
+	}
+
+	for _, template := range dynamicTemplates {
+		if len(template) != 1 {
+			return fmt.Errorf("unexpected number of dynamic template definitions found")
+		}
+
+		// there is just one dynamic template per object
+		templateName := ""
+		var rawContents any
+		for key, value := range template {
+			templateName = key
+			rawContents = value
+		}
+
+		if shouldSkipDynamicTemplate(templateName) {
+			continue
+		}
+
+		// logger.Debugf("Checking dynamic template for %q: %q", currentPath, templateName)
+		contents, ok := rawContents.(map[string]any)
+		if !ok {
+			return fmt.Errorf("unexpected dynamic template format found for %q", templateName)
+		}
+
+		fullRegex := false
+		if v, ok := contents["match_pattern"]; ok {
+			s, ok := v.(string)
+			if !ok {
+				return fmt.Errorf("invalid type for \"match_pattern\": %T", v)
+			}
+			if s == "regex" {
+				logger.Debugf("Use full regex in dynamic templates (match_pattern: regex)")
+				fullRegex = true
+			}
+		}
+
+		// matches with the current definitions and path
+		// https://www.elastic.co/guide/en/elasticsearch/reference/current/dynamic-templates.html
+		allMatched := true
+		for setting, value := range contents {
+			matched := true
+			switch setting {
+			case "mapping", "match_pattern":
+				// Do nothing
+			case "match":
+				name := fieldNameFromPath(currentPath)
+				// logger.Debugf("> Check match: %q (key %q)", currentPath, name)
+				values, err := parseSetting(value)
+				if err != nil {
+					logger.Warnf("failed to check match setting: %s", err)
+					return fmt.Errorf("failed to check match setting: %w", err)
+				}
+				if slices.Contains(values, name) {
+					logger.Warnf(">>>> no contained %s: %s", values, name)
+					continue
+				}
+
+				var matches bool
+				if fullRegex {
+					matches, err = stringMatchesPatterns(values, name)
+				} else {
+					matches, err = stringMatchesWildcards(values, name)
+				}
+				if err != nil {
+					return fmt.Errorf("failed to parse dynamic template %s: %w", templateName, err)
+				}
+
+				if !matches {
+					// logger.Debugf(">> Issue: not matches")
+					matched = false
+				}
+			case "unmatch":
+				name := fieldNameFromPath(currentPath)
+				// logger.Debugf("> Check unmatch: %q (key %q)", currentPath, name)
+				values, err := parseSetting(value)
+				if err != nil {
+					return fmt.Errorf("failed to check unmatch setting: %w", err)
+				}
+				if slices.Contains(values, name) {
+					matched = false
+					break
+				}
+
+				var matches bool
+				if fullRegex {
+					matches, err = stringMatchesPatterns(values, name)
+				} else {
+					matches, err = stringMatchesWildcards(values, name)
+				}
```

Comment:
Yeah, the docs seem to indicate that it only applies to `match`, but it would be weird that it is not possible to set this option for `unmatch`. Maybe `match_pattern` applies to both?

In any case, do we generate templates with these options?